src/managers/config.py

```diff
@@ -118,12 +128,22 @@ def port(self) -> int:
         Returns:
             Integer of port number
         """
+        # generates ports 39092, 39192, 39292 etc for listener auth if baseport=30000
+        if self.scope == "EXTRA":
+            return (
+                getattr(SECURITY_PROTOCOL_PORTS[self.auth_map], "client")
+                + self.baseport
+                + (self.extra_count * 100)
+            )
```

Comment:
(issue) Sorry, but I don't understand this part of the code. From the code it seems that we are offsetting of 100 based on the listener order. Are we thinking that if we would have the case:

`bar-{unit}:3000,foo-{unit}:3000` which needs to create `bar-0:39092` and `foo-0:39192`? My thought is that in this case the config should rather be `bar-{unit}:3000,foo-{unit}:3100`. The user should know that they should not use the same baseport for different extra listeners

Reply:
Hi sorry if the user provides the value 30000 twice for different hostnames that should raise an error. The base port numbers should be unique

Reply:
The error is raised in the validation step in the structured config already, so we're covered there.
I got rid of the `extra_count * 100` part as I realise it wasn't needed - [`1e4bb90` (#269)](https://github.com/canonical/kafka-operator/pull/269/commits/1e4bb90926bda3d68479a23ee1e0edf845a27625)